Test.LightweightIocContainer/IocValidatorTest.cs

```diff
@@ -59,7 +64,18 @@ private class InvalidTestInstaller : IIocInstaller
         public void TestValidate()
         {
             IocContainer iocContainer = new();
-            iocContainer.Install(new TestInstaller());
+            iocContainer.Install(new TestInstallerNoFactory());
+            
+            IocValidator validator = new(iocContainer);
+            
+            validator.Validate();
```

Comment:
should't this test fail from your perspective?

Reply:
Hm, didn't think about that - but yes, I think it should.
@SimonG96 what do you think?

Reply:
Okay, my bad. It actually DOES fail